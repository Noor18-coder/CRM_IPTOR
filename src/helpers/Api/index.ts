export * from './ApiRequest';
export * from './User';
export * from './CompanyInfo';
export * from './CurrencyInfo';
export * from './StagesInfo';
export * from './OpportunityType';
export * from './Attributes';
