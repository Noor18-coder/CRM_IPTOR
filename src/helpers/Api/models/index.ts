export * from './UserItem';
export * from './CompanyInfo';
export * from './ApiRequestControls';
export * from './Opportunity';
export * from './OpportunityDetails';
export * from './Users';
export * from './StageInfo';
export * from './Contacts';
export * from './Product';
export * from './Customer';
export * from './OpportunityTypes';
export * from './CustomerDetails';
export * from './AddOpportunity';
export * from './Items';
export * from './OpportunityAttributes';
export * from './Currency';
export * from './Attributes';
export * from './AddCustomer';
export * from './Countries';
export * from './ForeCastInfo';
