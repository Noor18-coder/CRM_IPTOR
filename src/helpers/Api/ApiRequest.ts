/**
 * Api Request Object
 */
import { ApiRequestControls } from './models';

export class ApiRequest<T> {
  IptorAPI: string;

  id: string;

  constructor(public method: string, public params?: T, public control?: ApiRequestControls) {
    this.IptorAPI = process.env.REACT_APP_IPTOR_API_VERSION ?? '';
    this.id = process.env.REACT_APP_IPTOR_API_ID ?? '';
  }
}
