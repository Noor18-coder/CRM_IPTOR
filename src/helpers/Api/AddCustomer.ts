import axios, { AxiosResponse } from 'axios';
import { get } from 'lodash';
import { AddBusinessPartnerDefaultParams, AddBusinessPartnerResponse, SaveUserDefinedFieldParam } from './models';
import { ApiRequest } from './ApiRequest';

export default class AddCustomerApi {
  /** API Method */
  private static apiMethod = 'crmBusinessPartner.add';

  private static updateApiMethod = 'crmBusinessPartner.update';

  private static saveAttributeValueMethod = 'mopAttribute.add';

  private static saveAttributeValueMethodFile = 'SRONAM';

  /**
   * Helper function to fetch Business Partner Info
   * API Method: crmBusinessPartner.add
   * */
  static async add(businessPartner: AddBusinessPartnerDefaultParams): Promise<AddBusinessPartnerResponse> {
    const requestData = new ApiRequest<AddBusinessPartnerDefaultParams>(this.apiMethod, businessPartner);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  static async update(businessPartner: AddBusinessPartnerDefaultParams): Promise<AddBusinessPartnerResponse> {
    const requestData = new ApiRequest<AddBusinessPartnerDefaultParams>(this.updateApiMethod, businessPartner);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  static async addAttributes(businessPartnerId: string, attributeType: string, attributeValue: string | number): Promise<AddBusinessPartnerResponse> {
    const params: SaveUserDefinedFieldParam = {
      attributeType,
      parentFile: this.saveAttributeValueMethodFile,
      parentId: businessPartnerId,
      attributeValue,
    };
    const requestData = new ApiRequest<SaveUserDefinedFieldParam>(this.saveAttributeValueMethod, params);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }
}
