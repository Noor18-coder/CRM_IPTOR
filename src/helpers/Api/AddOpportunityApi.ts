import axios, { AxiosResponse } from 'axios';
import { AddOpportunityDefaultParams, AddOpportunityResponse, SaveUserDefinedFieldParam, AddItemToOpportunityParams, AddItemToOpportunityResponse, AddCustomerContactParams, AddCustomerContactRequestParams } from './models';
import { ApiRequest } from './ApiRequest';
import { get } from 'lodash';

export default class AddOpportunityApi {
  /** API Method */
  private static apiMethod: string = 'mopOpportunity.add';
  private static saveAttributeValueMethod: string = 'mopAttribute.add';
  private static saveAttributeValueMethodFile: string = 'SROMOPH';
  private static saveItemMethod: string = 'mopItem.add';
  private static saveContactFileName: string = 'SROMOPCH';
  private static saveContactMethod: string = 'mopContact.add';

  /**
   * Helper function to fetch Business Partner Info
   * API Method: mopOpportunity.get
   * @param freeTextSearch search term
   * 
   */
  static async add(opportunity: AddOpportunityDefaultParams): Promise<AddOpportunityResponse> {
    const requestData = new ApiRequest<AddOpportunityDefaultParams>(this.apiMethod, opportunity);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  /**
   * Helper function to fetch Business Partner Info
   * API Method: mopOpportunity.get
   * @param freeTextSearch search term
   * 
   */
  static async addAttributes(opportunityId: string, attributeType: string, attributeValue: string | number): Promise<AddOpportunityResponse> {
    const params: SaveUserDefinedFieldParam = {
      attributeType: attributeType,
      parentFile: this.saveAttributeValueMethodFile,
      parentId: opportunityId,
      attributeValue: attributeValue
    }
    const requestData = new ApiRequest<SaveUserDefinedFieldParam>(this.saveAttributeValueMethod, params);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  /**
   * Helper function to fetch Business Partner Info
   * API Method: mopOpportunity.get
   * @param freeTextSearch search term
   * 
   */
  static async addItem(opportunityId: string, item: string, quantity: number, unit:string): Promise<AddItemToOpportunityResponse> {
    const params: AddItemToOpportunityParams = {
      parentFile: this.saveAttributeValueMethodFile,
      parentId: opportunityId,
      item: item,
      quantity: quantity,
      unit:unit
    }
    const requestData = new ApiRequest<AddItemToOpportunityParams>(this.saveItemMethod, params);
    const response = await axios.post<AddItemToOpportunityResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  /**
   * Helper function to fetch Business Partner Info
   * API Method: mopOpportunity.get
   * @param freeTextSearch search term
   * 
   */
   static async addContacts(params:AddCustomerContactParams): Promise<AxiosResponse> {
    const request:AddCustomerContactRequestParams  = {
      contactParentFile:this.saveAttributeValueMethodFile,
      contactParentId:params.contactPArentId,
      contactPerson:params.contactPerson,
      phone:params.phone,
      mobile:params.mobile,
      contactDC:'',
      whatsApp:'',
      linkedin:'',
      fax:'',
      email:params.email
    }

    const requestData = new ApiRequest<AddCustomerContactRequestParams>(this.saveContactMethod, request);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    console.log(response);
    return get(response, 'data');
  }
}
