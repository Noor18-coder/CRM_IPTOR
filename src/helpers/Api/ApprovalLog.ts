import axios, { AxiosResponse } from 'axios';
import { get } from 'lodash';
import { ApprovalLogAddRequestParams, ApprovalLogAddResponse, DeleteApprovalLogsParams } from './models';
import { ApiRequest } from './ApiRequest';

export class ApprovalLog {
  /** API Method */
  private static apiMethod = 'crmOpportunityApprovalLog.add';

  private static deleteApprovalLogs = 'crmOpportunityApprovalLog.delete';

  /**
   * Helper function to fetch sales order list
   * @param params user id
   * @returns information about user
   */
  static async addApprovalLog(params: ApprovalLogAddRequestParams): Promise<ApprovalLogAddResponse> {
    const requestData = new ApiRequest<ApprovalLogAddRequestParams>(this.apiMethod, params);
    const response = await axios.post<ApprovalLogAddResponse>('/api/service', requestData);
    return get<AxiosResponse<ApprovalLogAddResponse>, 'data'>(response, 'data');
  }

  static async deleteOpportunityApprovalLogs(params: DeleteApprovalLogsParams): Promise<ApprovalLogAddResponse> {
    const requestData = new ApiRequest<DeleteApprovalLogsParams>(this.deleteApprovalLogs, params);
    const response = await axios.post<ApprovalLogAddResponse>('/api/service', requestData);
    return get(response, 'data.data.items', {});
  }
}
