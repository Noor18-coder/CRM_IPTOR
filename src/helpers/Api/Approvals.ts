import axios, { AxiosResponse } from 'axios';
import { get } from 'lodash';
import * as apiModels from './models';
import { ApiRequest } from './ApiRequest';

export class ApprovalInfo {
  /** API Method */
  private static allApprovalsMethod = 'crmOpportunityApprovalLogs.get';

  private static allNotesMethod = 'mopNotes.get';

  private static notesParentFile = 'SROMOPAL';

  static async getApprovalLogs(opportunityId: string): Promise<apiModels.ApprovalLogsDefault[]> {
    const requestData = new ApiRequest<apiModels.ApprovalLogsParams>(this.allApprovalsMethod, { opportunityId });
    const response = await axios.post<apiModels.ApprovalLogsResponse>('/api/service', requestData);
    return get<AxiosResponse<apiModels.ApprovalLogsResponse>, 'data', 'data', 'items', apiModels.ApprovalLogsDefault[]>(
      response,
      ['data', 'data', 'items'],
      []
    );
  }

  static async getAllNotes(parentId: string): Promise<apiModels.AllNotesDefault> {
    const requestData = new ApiRequest<apiModels.AllNotesParams>(this.allNotesMethod, { parentFile: this.notesParentFile, parentId });
    const response = await axios.post<apiModels.AllNotesResponse>('/api/service', requestData);
    const defaultData = {} as apiModels.AllNotesDefault;
    return get<AxiosResponse<apiModels.AllNotesResponse>, 'data', 'data', 'items', apiModels.AllNotesDefault>(
      response,
      ['data', 'data', 'items'],
      defaultData
    );
  }
}
