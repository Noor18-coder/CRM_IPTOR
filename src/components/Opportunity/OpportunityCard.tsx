import { useSelector } from 'react-redux';
import { getCurrencySymbol, getQuarterOfYearFromDate } from '../../helpers/utilities/lib';
import { OpportunityListItem } from '../../helpers/Api/models';
// import ImageConfig from '../../config/ImageConfig';
import { AppState } from '../../store/store';
import { UsersData } from '../../store/Users/Types';

interface OppProps {
  opportunity: OpportunityListItem;
  name: string;
}

const OpportunityCard: React.FC<React.PropsWithChildren<OppProps>> = (props) => {
  const oppState: AppState = useSelector((appState: AppState) => appState);
  const usersData: UsersData = useSelector((state: AppState) => state.users);
  const {
    opportunity: { desc, opportunityId, name, stage, oppRecordType, endDate, forecastCategory, estValueSys, userId },
  } = props;

  const getName = (str: string) => {
    const userObj = usersData.users.find((obj) => obj.user === str);
    return userObj?.description;
  };

  const formatDealSizeValue = () => {
    const CurrencySymbol = oppState.enviornmentConfigs.defaultOpprtunityInfo.currencyLDA
      ? getCurrencySymbol(oppState.enviornmentConfigs.defaultOpprtunityInfo.currencyLDA)
      : '';
    const renderValue = `${CurrencySymbol} ${estValueSys}`;
    return renderValue;
  };

  return (
    <>
      <div className="d-flex justify-content-between title-row">
        <div className="lft-col">
          {desc}
          <span>
            {opportunityId} | {name}
          </span>
        </div>
        <div className="rgt-col">
          <span className="danger">{stage}</span>
        </div>
      </div>
      <div className="d-flex justify-content-between owner-row">
        <div className="lft-col">
          Record Type
          <span>{oppRecordType}</span>
        </div>
        <div className="rgt-col">
          Close Quarter
          <span>{getQuarterOfYearFromDate(`${endDate}`)}</span>
        </div>
      </div>

      <div className="d-flex justify-content-between qtr-details-row">
        <div className="lft-col">
          Owner
          <span>{getName(userId)}</span>
        </div>
        <div className="rgt-col">
          Forecast
          <span>{forecastCategory}</span>
        </div>
      </div>

      <div className="d-flex justify-content-between location-row">
        <div className="lft-col">{formatDealSizeValue()}</div>
        {/* Commented non-working button <div className="rgt-col">
          <img src={ImageConfig.NAV_MORE_DOTS} alt="More" title="More" />
        </div> */}
      </div>
    </>
  );
};

export default OpportunityCard;
