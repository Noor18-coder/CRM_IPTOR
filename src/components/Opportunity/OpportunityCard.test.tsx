import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import OpportunityCard from './OpportunityCard';
import StoreMock from '../../mocks/Store.mock';
import { OpportunityListItem } from '../../helpers/Api/models';

const store = StoreMock.createInitialStore();
const opportunity = {} as OpportunityListItem;

describe('[Opportunity] OpportunityCard', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <OpportunityCard opportunity={opportunity} name="hello" />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
