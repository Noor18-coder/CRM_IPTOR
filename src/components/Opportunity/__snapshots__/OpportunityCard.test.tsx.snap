// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Opportunity] OpportunityCard should renders correctly 1`] = `
Array [
  <div
    className="d-flex justify-content-between title-row"
  >
    <div
      className="lft-col"
    >
      <span>
         | 
      </span>
    </div>
    <div
      className="rgt-col"
    >
      <span
        className="danger"
      />
    </div>
  </div>,
  <div
    className="d-flex justify-content-between owner-row"
  >
    <div
      className="lft-col"
    >
      Record Type
      <span />
    </div>
    <div
      className="rgt-col"
    >
      Close Quarter
      <span>
        -
      </span>
    </div>
  </div>,
  <div
    className="d-flex justify-content-between qtr-details-row"
  >
    <div
      className="lft-col"
    >
      Owner
      <span />
    </div>
    <div
      className="rgt-col"
    >
      Forecast
      <span />
    </div>
  </div>,
  <div
    className="d-flex justify-content-between location-row"
  >
    <div
      className="lft-col"
    >
       undefined
    </div>
  </div>,
]
`;
